packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -132,95 +163,133 @@ - (void)initialize:(FlutterError *__autoreleasing *)error {
   upgradeAudioSessionCategory(AVAudioSessionCategoryPlayback, 0, 0);
 #endif
 
-  [self.playersByTextureId
-      enumerateKeysAndObjectsUsingBlock:^(NSNumber *textureId, FVPVideoPlayer *player, BOOL *stop) {
-        [self.registry unregisterTexture:textureId.unsignedIntegerValue];
-        [player dispose];
-      }];
-  [self.playersByTextureId removeAllObjects];
+  [self.playersById.allValues makeObjectsPerformSelector:@selector(dispose)];
+  [self.playersById removeAllObjects];
 }
 
 - (nullable NSNumber *)createWithOptions:(nonnull FVPCreationOptions *)options
                                    error:(FlutterError **)error {
+  BOOL textureBased = options.viewType == FVPPlatformVideoViewTypeTextureView;
+
+  @try {
+    FVPVideoPlayer *player;
+    if (textureBased) {
+      player = [self createTexturePlayerWithOptions:options];
+    } else {
+      player = [self createPlatformViewPlayerWithOptions:options];
+    }
```

Comment:
Optional:

```objc
FVPVideoPlayer *player = textureBased
    ? [self createTexturePlayerWithOptions:options]
    : [self createPlatformViewPlayerWithOptions:options];
```

(This keeps the declaration and initialization in the same statement, making it clearer that it's always initialized.)